stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/composer/MessageComposerDefaultCenterContent.kt

```diff
@@ -105,6 +105,8 @@ public class MessageComposerDefaultCenterContent : FrameLayout, MessageComposerC
             val newValue = state.inputValue
             if (newValue != currentValue) {
                 setText(state.inputValue)
+                //placing cursor at the end of the text
+                setSelection(length())
```

Comment:
How does this work if we write something in the middle of the text? The value changes, but we don't want to push the cursor at the end?

We have a fix for this in Compose where we check the previous position and only push to the end if the user was already there.

Reply:
The condition checks if the current input in `MessageInputState` is different from the text displayed in `EditText`. It moves the cursor only if there is a difference. That's why in the case you describe the cursor will not be moved.

Reply:
Tested and seems to be working! :]